front/src/applications/stdcm/components/StdcmLoader.tsx

```diff
@@ -1,42 +1,57 @@
-import { forwardRef } from 'react';
+import { useEffect } from 'react';
 
 import { Button } from '@osrd-project/ui-core';
 import { useTranslation } from 'react-i18next';
 
-import stdcmLoaderImg from 'assets/pictures/views/stdcm_loader.jpg';
+const LOADER_OFFSET = 32;
 
 type StdcmLoaderProps = {
   cancelStdcmRequest: () => void;
 };
 
-const StdcmLoader = forwardRef(
-  ({ cancelStdcmRequest }: StdcmLoaderProps, ref: React.Ref<HTMLDivElement>) => {
-    const { t } = useTranslation('stdcm');
-    return (
-      <div className="stdcm-loader-background">
-        <div
-          ref={ref}
-          className="stdcm-loader d-flex flex-column justify-content-center align-items-center"
-        >
-          <div className="stdcm-loader__wrapper">
-            <h1>{t('simulation.calculatingSimulation')}</h1>
-            <p>{t('simulation.averageRequestTime')}</p>
-          </div>
-          <div className="stdcm-loader__cancel-btn">
-            <Button
-              data-testid="cancel-simulation-button"
-              variant="Cancel"
-              label={t('simulation.stopCalculation')}
-              size="small"
-              onClick={cancelStdcmRequest}
-            />
-          </div>
-          <img src={stdcmLoaderImg} alt={t('simulation.pendingSimulation')} />
-          <p className="stdcm-loader__img-signature">{t('loaderImageLegend')}</p>
+const StdcmLoader = ({ cancelStdcmRequest }: StdcmLoaderProps) => {
+  const { t } = useTranslation('stdcm');
+
+  useEffect(() => {
+    // Depending on the scroll change the position of the loader between fixed and sticky
+    const handleScroll = () => {
+      const element = document.querySelector('.stdcm-loader') as HTMLElement;
+      const scrollTop = window.scrollY;
+
+      if (scrollTop > window.innerHeight - LOADER_OFFSET) {
```

Comment:
Do we need to have a constant for this? Could we use a ref and find the height of the HTML element dynamically instead?

Reply:
The threshold where we want to change is position depends of this offset. Also I made a mistake, it's actually not 32 but 32 * 2 (the offset bottom when it's sticky and the offset top when it's fixed). Do you see a patern I'm missing depending of the component's height ?

Reply:
I still kept a few `const` like this one. I'm not sure if having a bunch of ref to pass in props would be a better solution. What do you think ?

Reply:
In general I dislike hardcoding style-related things in JS, when it really should just be in the CSS. The reason is that (1) it's fragile (depends on font size, padding, margin, general layout of the page, etc) and (2) it duplicates the information in two spots (CSS and JS).